Multisig/UI/Settings/OwnerKeyManagement/AddKeyAsNewOwner/RemoveOwnerFlow.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  RemoveOwnerFlow.swift
+//  Multisig
+//
+//  Created by Moaaz on 5/11/22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+
+class RemoveOwnerFlow: UIFlow {
+    var factory: RemoveOwnerFlowFactory
+    var safe: Safe
+    var ownerToRemove: Address?
+    var prevOwner: Address?
+    var newConfirmations: Int?
+    var removeOwnerTransactionDetails: SCGModels.TransactionDetails?
+
+    internal init(owner: Address, safe: Safe, factory: RemoveOwnerFlowFactory = .init(), navigationController: UINavigationController, completion: @escaping (_ success: Bool) -> Void) {
+        self.factory = factory
+        self.safe = safe
+        self.ownerToRemove = owner
+        super.init(navigationController: navigationController, completion: completion)
+    }
+
+    override func start() {
+        confirmations()
+    }
+
+    func confirmations() {
+        let confirmationsVC = factory.confirmations(
+            step: 1,
+            maxSteps: 2,
+            safe: safe
+        ) { [unowned self] newConfirmations in
+            self.newConfirmations = newConfirmations
+            review()
+        }
+        show(confirmationsVC)
+    }
+
+    func review() {
+        assert(ownerToRemove != nil)
+        assert(newConfirmations != nil)
+
+        let reviewVC = factory.review(
+            step: 2,
+            maxSteps: 2,
+            safe: safe,
+            owner: ownerToRemove!,
+            prevOwner: prevOwner, newThreshold: newConfirmations!) { [unowned self] txDetails in
+                removeOwnerTransactionDetails = txDetails
+                success()
+            }
+        show(reviewVC)
+    }
+
+    func success() {
+        assert(removeOwnerTransactionDetails != nil)
+        let successVC = factory.success { [unowned self] showTxDetails in
+            if showTxDetails {
+                NotificationCenter.default.post(
+                    name: .initiateTxNotificationReceived,
+                    object: self,
+                    userInfo: ["transactionDetails": removeOwnerTransactionDetails!])
+            }
+            stop(success: !showTxDetails)
+        }
+        show(successVC)
+    }
+}
+
+class RemoveOwnerFlowFactory {
+    func confirmations(step: Int, maxSteps: Int, safe: Safe, completion: @escaping (_ newConfirmations: Int) -> Void) -> EditConfirmationsViewController {
+        let confirmationsVC = EditConfirmationsViewController()
+        confirmationsVC.confirmations = Int(safe.threshold ?? 0)
+        confirmationsVC.minConfirmations = 1
+        confirmationsVC.maxConfirmations = max(1, (safe.ownersInfo ?? []).count) - 1
+        confirmationsVC.stepNumber = step
+        confirmationsVC.maxSteps = maxSteps
+        confirmationsVC.trackingEvent = .addAsOwnerChangeConfirmations
```

Comment:
is tracking right here?

Is this code repeating something else or is it OK? Do we need to generalize "factory" classes?